### PR TITLE
Pull Request of Skin Disease Prediction

### PR DESCRIPTION
**Pull request by keshripritesh**
---

### 🚀 Issue: Add New Project - Skin Disease Predictor

**Project Title**: Skin Disease Predictor

**Description**:
This project is focused on predicting various skin diseases using deep learning, specifically **Convolutional Neural Networks (CNN)** with **Transfer Learning** techniques. The model is trained on a publicly available dataset from **DermNet**, and aims to classify different types of skin conditions efficiently. This solution has potential use in telemedicine, remote health diagnostics, and dermatological research.

---

### 📂 Repository Structure

```
skin-disease-predictor/
├── dataset/                  # Image dataset (not uploaded directly due to size)
├── model/                   # Pretrained model and saved weights
├── notebook/                # Jupyter Notebook for training and testing
├── utils/                   # Preprocessing and helper scripts
├── requirements.txt         # All dependencies
└── README.md                # Complete documentation
```

---

### 🛠 Tech Stack

* Python
* TensorFlow
* Keras
* NumPy
* CNN (Transfer Learning with VGG16 / ResNet)

---

### 📦 Setup Instructions

1. Clone the repo and install required packages:

```bash
pip install keras
pip install tensorflow
pip install numpy
pip install glob
```

2. Download the dataset from Kaggle: [[DermNet Dataset](https://www.kaggle.com/datasets/shubhamgoel27/dermnet)](https://www.kaggle.com/datasets/shubhamgoel27/dermnet)

3. Run the Jupyter Notebook provided to train and evaluate the model.

---

---

### 💡 Why This Project?

* Addresses a real-world healthcare challenge using AI.
* Great for contributors interested in **Medical AI**, **Computer Vision**, and **Transfer Learning**.
* Open for enhancement: add more skin diseases, improve accuracy, integrate into web apps, etc.

---

### ✅ Goals for SSOC Contributors

* Refactor code for modularity and scalability
* Add a Streamlit-based web interface
* Perform model evaluation using more metrics (AUC, F1-Score, etc.)
* Optimize the model using MobileNet for edge deployment
* Expand dataset preprocessing pipeline
* Add Docker support

---

### 🔖 Tags

`#deep-learning` `#cnn` `#skin-disease` `#healthcare` `#keras` `#tensorflow` `#computer-vision` `#ssoc`

---

### CODE REVIEW

**Review by PRIYANSHU2026 — APPROVED** [review]
Good work @keshripritesh
